crates/full-node/sov-sequencer/src/preferred/inner.rs

```diff
@@ -661,6 +667,25 @@ where
 
         Ok(())
     }
+
+    pub(crate) async fn update_api_ledger(&self, info: &StateUpdateInfo<S::Storage>) {
+        let start = std::time::Instant::now();
+        tracing::trace!(
+            slot_number = %info.slot_number,
+            latest_finalized_slot_number = %info.latest_finalized_slot_number,
+            "Starting LedgerAPI storage update");
+        self.api_ledger_db
+            .replace_reader(info.ledger_reader.clone());
```

Comment:
I've been playing with this in `https://github.com/Sovereign-Labs/sovereign-sdk/pull/1478

Could you please add trace between replace reader and before send_notifications? That would be helpful.

Reply:
This is non-blocking, obviously.